ambry-cloud/src/main/java/com/github/ambry/cloud/CloudBlobStore.java

```diff
@@ -1061,6 +1512,39 @@ public void appendFrom(ReadableByteChannel channel, long size) throws StoreExcep
         throw new StoreException(e, StoreErrorCodes.IOError);
       }
     }
+
+    /**
+     * Appends the channel to the underlying write interface. Writes "size" number of bytes to the interface.
+     * @param channel The channel from which data needs to be written from
+     * @param size The amount of data in bytes to be written from the channel
+     * @return a {@link CompletableFuture} that will eventually complete successfully once the bytes are written to the
+     *         write interface or an exception if an error occurs.
+     */
+    public CompletableFuture<Void> appendAsyncFrom(ReadableByteChannel channel, long size) {
+      // Upload the blob corresponding to the current message index
+      MessageInfo messageInfo = messageInfoList.get(messageIndex++);
+      if (messageInfo.getSize() != size) {
+        return FutureUtils.completedExceptionally(
```

Comment:
let's formalize IllegalStateException and IllegalArgumentException, do we want to wrap them in the CompleteFuture?

Reply:
Made changes to be consistent and not wrap IllegalStateException and IllegalArgumentException in CompletionException since they are runtime exceptions.